signers/signer-solana/src/utils/simulate.ts

```diff
@@ -49,23 +46,37 @@ function getSimulationError(
    * trying to detect common errors (e.g. insufficient fund or slippage error)
    * We could probably remove this code after upgrading solana/web3 lib to v2
    */
-  if (
-    typeof error === 'object' &&
-    error?.InstructionError?.[0] === INSTRUCTION_INDEX &&
-    typeof error?.InstructionError?.[1] === 'object'
-  ) {
-    if (
-      error?.InstructionError?.[1]?.Custom === INSUFFICIENT_FUNDS_ERROR_CODE
-    ) {
-      const insufficentLamportErrorMessage = logs?.find((log) =>
-        log.toLowerCase()?.includes('insufficient lamports')
-      );
+  if (typeof error === 'string') {
```

Comment:
NIT: You could update this part of code to avoid potential index out of range errors and make it a bit more clean.

```ts
if (typeof error === 'string') {
  message = error === ACCOUNT_NOT_FOUND_ERROR
    ? 'Attempt to debit an account but found no record of a prior credit.'
    : error;
} else if (Array.isArray(error?.InstructionError) && error.InstructionError.length > 1) {
  const instructionError = error.InstructionError[1];

  if (typeof instructionError === 'object') {
    switch (instructionError.Custom) {
      case INSUFFICIENT_FUNDS_ERROR_CODE:
        const insufficientLamportErrorMessage = logs?.find((log) =>
          log.toLowerCase()?.includes('insufficient lamports')
        );
        message = insufficientLamportErrorMessage || 'Insufficient funds';
        break;
      case SLIPPAGE_ERROR_CODE:
        message = 'Slippage error';
        break;
      default:
        break;
    }
  } else if (instructionError === PROGRAM_FAILED_TO_COMPLETE_ERROR) {
    message = 'Program failed to complete';
  }
} else if (error?.InsufficientFundsForRent) {
  message = 'Transaction results in an account with insufficient funds for rent.';
}

```